doc/smart-switch/BFD/SmartSwitchDpuLivenessUsingBfd.md

```diff
@@ -0,0 +1,265 @@
+    pair.
+
+## 1.2 Scale Requirements
+
+- 64 BFD active sessions from NPU to each DPU in a T1 cluster.
+
+- 8 BFD passive sessions from DPU to all NPUs in a T1 cluster.
+  
+Note: Scale requirements will  double when BFD sessions need to be created for both IPv4 and IPv6.
+
+## 1.3 Default values
+
+- BFD TX_INTERVAL -- 100 msec(might change)
+
+- BFD_RX_INTERVAL - 100 msec(might change)
+
+- BFD session detect time multiplier - 3
+
+- BFD session type: Active on NPU and Passive on DPU
+
+- BFD mode: Multihop mode for sessions between NPU and remote DPU
+
+# 2. Modules Design
+## 2.1 APP_DB changes
+Existing VNET_ROUTE_TUNNEL_TABLE is updated to include BFD Rx and Tx timers. Overlay ECMP feature uses the default of one second and Smartswitch can have aggressive timers due to less number of BFD sessions.
+
+Existing
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    "mac_address":{{mac_address1},{mac_address2},...} (OPTIONAL)  
+    "vni": {{vni1},{vni2},...} (OPTIONAL) 
+    "weight": {{w1},{w2},...} (OPTIONAL) 
+    “profile”: {{profile_name}} (OPTIONAL) 
+```
+
+Proposed changes
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    "mac_address":{{mac_address1},{mac_address2},...} (OPTIONAL)  
+    "vni": {{vni1},{vni2},...} (OPTIONAL) 
+    "weight": {{w1},{w2},...} (OPTIONAL) 
+    “profile”: {{profile_name}} (OPTIONAL) 
+    "rx_monitor_timer": {time in milliseconds}
+    "tx_monitor_timer": {time in milliseconds}
+```
+
+## 2.2 SmartSwitch NPU-DPU BFD sessions
+
+In a Smartswitch deployment, BFD sessions are created between NPU and
+DPUs. Each T1 cluster has 8 T1s and 8 DPUs in each T1. To detect the
+liveness of each DPU in the cluster, NPU creates Active BFD sessions to
+all 64 DPUs and each DPU creates passive BFD sessions to all NPUs in the
+cluster. When BFD keepalives are not detected for certain interval, BFD
+sessions are marked as down and corresponding action is performed.
+Multihop BFD sessions are created between NPU and a remote DPU.
+
+<img src="images/NPU_DPU_BFD_sessions.png">
+
+## 2.3 BFD Active-Passive mode
+
+HA manager in NPU creates VNET_ROUTE_TUNNEL_TABLE which will trigger bfdOrch to create BFD Active session with local and peer information. NPU supports BFD hardware offload, so hardware starts
+generating BFD packets to the peer. HA manager also creates another
+BFD_SESSION table in DPU with local DPU PA and peer NPU PAs. BfdOrch checks for SAI object platform capability for h/w offload. In DPU, if h/w offload is not supported, FRR is configured to trigger BFD sessions. BfdOrch creates SOFTWARE_BFD_SESSION_TABLE entry in STATE_DB with 'passive' mode and BgpCfgd checks for the entries in the STATE_DB and configures BFD passive sessions in

```

Comment:
HA manager also creates another BFD_SESSION table in DPU

to 

HA manager also creates BFD passive sessions in DPU using `BFD_SESSION` table